packages/sdk-react-hooks/src/hooks/msg_channel.tsx

```diff
@@ -0,0 +1,27 @@
+import { useContext, useEffect, useState } from "react";
+import { Atm0sMediaContext } from "../provider";
+import {
+  MessageChannelConfig,
+  RoomMessageChannel,
+} from "@atm0s-media-sdk/core";
+
+export function useMessageChannel(
+  key: string,
+  callback: (event: {
+    key: string;
+    peer: string;
+    message: Uint8Array | string;
+  }) => void,
+  config?: MessageChannelConfig
+) {
+  const ctx = useContext(Atm0sMediaContext);
+  const [channel, setChannel] = useState<RoomMessageChannel | null>(null);
+  useEffect(() => {
```

Comment:
I think we need a function same with ctx.getOrCreatePublisher. Without it we will leak datachannel each time useEffect re-run

Reply:
instead of keeping a registry of msg channels in the react lib like the publishers, I've updated the core sdk to natively track the channels creation

Reply:
@luongngocminh It look good. Can you create same mechanism with publishers after this PR is merged?

Reply:
@giangndm do you want to move the publisher class into the core sdk? Because now it's only available in the react hooks side